ecs-cli/modules/commands/regcreds/regcreds_command.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package regcredsCommand
+
+import (
+	ecscli "github.com/aws/amazon-ecs-cli/ecs-cli/modules"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/regcreds"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/urfave/cli"
+)
+
+// RegistryCredsCommand provides a list of commands that facilitate use of private registry credentials with ECS.
+func RegistryCredsCommand() cli.Command {
+	return cli.Command{
+		Name:   "registry-creds",
+		Usage:  "Facilitates the creation and use of private registry credentials within ECS.",
+		Before: ecscli.BeforeApp,
+		Flags:  flags.OptionalRegionAndProfileFlags(),
+		Subcommands: []cli.Command{
+			upCommand(),
+		},
+	}
+}
+
+func upCommand() cli.Command {
+	return cli.Command{
+		Name:         "up",
+		Usage:        "Generates AWS Secrets Manager secrets and an IAM Task Execution Role for use in an ECS Task Definition.",
+		Action:       regcreds.Up,
+		Flags:        regcredsUpFlags(),
+		OnUsageError: flags.UsageErrorFactory("up"),
+	}
+}
+
+// TODO: add rest of flags as functionality implemented
```

Comment:
I realized that this probably should have been mentioned in the design, but 1) are we expecting creds to only be passed in via a file? and 2) this would presumably be a required parameter. Assuming the answer to both is yes, I am wondering if we could take this as an argument (like we do with `compose run <command override>`)rather than a flag, which would eliminate the stickiness of having a flag name that collides with an existing one.

Reply:
I wouldn't worry about the flag name colliding or anything; this is a separate command are re-use is fine. I do really like the idea of making the file an argument though! +1 for that

Reply:
1. Correct, cred info only passed in via file, which is required.
2. Hmm I had not considered making file name an argument! On the surface, sounds like this would make things simpler, but everywhere else we accept a file name we use a flag, so it feels inconsistent to use an arg in this specific case? We also, in general, use args (vs. flags) a lot less in the ECS CLI, so it doesn't feel as intuitive (example: I was not aware of the `compose run <override>` functionality until we started refactoring `run`). But! Those hunches could be based on incomplete/biased understanding of what others are used to doing. Let me poke around and try to convince myself that [arg] is the way to go...